### ISSUE TITLE
Error while importing preset on Windows 10 - darktable 3.0.1

### ISSUE THREAD

**Issue by jkaipa** [darktable-org/darktable#4690]
This is the exception received while importing preset into darktable.
```
this is darktable 3.0.1 reporting an exception:

-------------------

Error occurred on Saturday, April 11, 2020 at 20:06:16.

darktable.exe caused an Access Violation at location 00007FFA9F04D501 in module msvcrt.dll Reading from location 0000000000000000.

AddrPC           Params
00007FFA9F04D501 000000001BCFEEF0 0000000000000004 0000000000000000  msvcrt.dll!strlen
000000006366F389 000000000000007F 0000000000000010 000000001B7DFC00  libdarktable.dll!dt_presets_import_from_file  [D:/build/darktable/src/common/presets.c @ 247]
0000000063749C47 0000000000000000 0000000000000000 0000000100000292  libdarktable.dll!import_preset  [D:/build/darktable/src/gui/preferences.c @ 1283]
0000000067F06E66 0000000000CC0020 00000000FFFFFFFF 0000000000000000  libgobject-2.0-0.dll!g_closure_invoke
0000000067F24713 000000001B8166C0 00000000677479E3 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 000000000061D928 00000000679A53CA 00000000035D6070  libgobject-2.0-0.dll!g_signal_emit
0000000067747A82 0000000000000003 0000CE5659E90D0D 00000000035D6001  libgtk-3-0.dll!gtk_button_clicked
0000000067747AD6 0000000017ADDE50 000000006C276488 0000000000000000  libgtk-3-0.dll!gtk_button_clicked
0000000067F06D77 0000001C00000000 0000000D00000011 000000000000000D  libgobject-2.0-0.dll!g_closure_invoke
0000000067F24713 000000001B8166C0 0000000068DD40D2 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 0000000000000000 0000000000000001 0000000000000002  libgobject-2.0-0.dll!g_signal_emit
0000000067745DC6 0000000000000037 00000000FFFFFFFF 0000000000000000  libgtk-3-0.dll!gtk_builder_get_application
0000000067F06E66 0000000000000000 0000000000000000 0000000000000000  libgobject-2.0-0.dll!g_closure_invoke
0000000067F24713 000000001B7FD840 0000000017AE0EA0 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 0000000017ADDE50 00007FFAA0EDFC11 0000000000000000  libgobject-2.0-0.dll!g_signal_emit
000000006780FD6D 0000000000000000 00000000649E5B0D 0000000000000001  libgtk-3-0.dll!gtk_gesture_long_press_new
0000000067F097CE 0000000004617000 000000000194E0A0 0000000000000000  libgobject-2.0-0.dll!g_cclosure_marshal_VOID__BOXEDv
0000000067F06D77 000000000061E270 00000000000000A0 0000000000000001  libgobject-2.0-0.dll!g_closure_invoke
0000000067F24713 000000001B7FD840 000000160000003C 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 0000000000000001 0000000000000000 000000001B8165D0  libgobject-2.0-0.dll!g_signal_emit
000000006780CD59 000000000194E140 000000001B59F8E0 0000000000000002  libgtk-3-0.dll!gtk_glade_catalog_init
000000006780E2F3 0000000000000021 0000000000000000 0000A97D65F50B7F  libgtk-3-0.dll!gtk_gesture_handles_sequence
00000000678113B5 000000001B7FD840 000000001B59F8E0 0000000000000002  libgtk-3-0.dll!gtk_gesture_rotate_get_angle_delta
00000000677D7D1D 0000000000000001 00000000000000A0 0000000000000001  libgtk-3-0.dll!gtk_event_controller_handle_event
000000006799F4FC 00000000019BBF20 00000342649456D6 0000000000000000  libgtk-3-0.dll!gtk_requisition_free
0000000067702D05 0000000000000001 0000000000000000 0000000017AED160  libgtk-3-0.dll!0x2d05
0000000067F06D77 0000000000000000 0000000000000000 0000000005275EA0  libgobject-2.0-0.dll!g_closure_invoke
0000000067F23C09 000000001B8166C0 000000000194E0A0 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 0000000001A359E0 0000000000000008 0000000000000000  libgobject-2.0-0.dll!g_signal_emit
00000000679A0F5F 0000000005275F90 000000000194E0A0 0000000001A359E0  libgtk-3-0.dll!gtk_requisition_copy
00000000678596A7 0000000000000000 00000000649456D6 0000000000000000  libgtk-3-0.dll!gtk_lock_button_set_permission
000000006785BA1B 000000000061E600 000000000199AC10 0000000000000002  libgtk-3-0.dll!gtk_main_do_event
000000006C1CF0AA 0000000000000000 00000000649F4500 00000000019922C0  libgdk-3-0.dll!gdk_event_free
000000006C204D9F 000000000061E5F8 0000000000000000 0000000001991340  libgdk-3-0.dll!gdk_win32_drag_context_get_type
00000000649F7BA3 0000000000000578 0000000000000010 0000000000000000  libglib-2.0-0.dll!g_main_context_dispatch
00000000649F7DD8 00000000638FABA0 0000000067F0B53C 0000000001991340  libglib-2.0-0.dll!g_main_context_dispatch
00000000649F8254 000000001B8166C0 0000000067F24B48 0000000005275F90  libglib-2.0-0.dll!g_main_loop_run
00000000677BD2A2 0000000005275F90 0000000000000018 00000000638A1BCB  libgtk-3-0.dll!gtk_dialog_run
0000000063750AF7 000000000061E998 0000000000000000 3FF0000000000000  libdarktable.dll!dt_gui_preferences_show  [D:/build/darktable/src/gui/preferences.c @ 315]
0000000067F06E66 0000000000CC0020 00000000FFFFFFFF 0000000000000000  libgobject-2.0-0.dll!g_closure_invoke
0000000067F24713 00000000176D19F0 00000000677479FC 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 000000000061EE98 00000000679A53CA 00000000035D5B20  libgobject-2.0-0.dll!g_signal_emit
0000000067747A82 0000000000000003 0000CE5659E9389D 00000000035D5B01  libgtk-3-0.dll!gtk_button_clicked
0000000067747AD6 0000000017CD1630 000000006C276488 0000000000000000  libgtk-3-0.dll!gtk_button_clicked
0000000067F06E66 0000001C00000000 0000000D00000011 000000000000000D  libgobject-2.0-0.dll!g_closure_invoke
0000000067F24713 00000000176D19F0 0000000068DD40D2 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 0000000000000000 0000000000000001 0000000000000002  libgobject-2.0-0.dll!g_signal_emit
0000000067745DC6 0000000000000037 00000000FFFFFFFF 0000000000000000  libgtk-3-0.dll!gtk_builder_get_application
0000000067F06E66 0000000000000000 00000000008C061B 0000000017AF84B0  libgobject-2.0-0.dll!g_closure_invoke
0000000067F24713 000000001784D820 0000000017D2E780 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 0000000017CD1630 00007FFAA0EDFC11 0000000000000000  libgobject-2.0-0.dll!g_signal_emit
000000006780FD6D 0000000000000000 00000000649E5B0D 0000000000000001  libgtk-3-0.dll!gtk_gesture_long_press_new
0000000067F097CE 0000000004617000 000000000194E0A0 0000000000000000  libgobject-2.0-0.dll!g_cclosure_marshal_VOID__BOXEDv
0000000067F06E66 000000000061F7E0 00000000000000A0 0000000000000001  libgobject-2.0-0.dll!g_closure_invoke
0000000067F24713 000000001784D820 0000001A00000016 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 0000000000000001 0000000000000000 00000000176D1900  libgobject-2.0-0.dll!g_signal_emit
000000006780CD59 000000000194E140 0000000017952E60 0000000000000002  libgtk-3-0.dll!gtk_glade_catalog_init
000000006780E2F3 0000000000000021 0000000000000000 0000A97D65F521EF  libgtk-3-0.dll!gtk_gesture_handles_sequence
00000000678113B5 000000001784D820 0000000017952E60 0000000000000002  libgtk-3-0.dll!gtk_gesture_rotate_get_angle_delta
00000000677D7D1D 0000000000000001 00000000000000A0 0000000000000001  libgtk-3-0.dll!gtk_event_controller_handle_event
000000006799F4FC 0000000017C628B0 00000001649456D6 0000000000000000  libgtk-3-0.dll!gtk_requisition_free
0000000067702D05 0000000000000001 0000000000000000 0000000017D2E740  libgtk-3-0.dll!0x2d05
0000000067F06E66 0000000000000000 0000000000000000 0000000005DB1C30  libgobject-2.0-0.dll!g_closure_invoke
0000000067F23C09 00000000176D19F0 000000000194E0A0 0000000000000000  libgobject-2.0-0.dll!g_signal_emit_valist
0000000067F24B48 0000000001A359E0 0000000000000008 0000000000000000  libgobject-2.0-0.dll!g_signal_emit
00000000679A0F5F 0000000000000000 000000000194E0A0 0000000001A359E0  libgtk-3-0.dll!gtk_requisition_copy
00000000678596A7 0000000000000000 00000000649456D6 0000000000000000  libgtk-3-0.dll!gtk_lock_button_set_permission
000000006785BA1B 000000000061FB70 000000000199AC10 0000000000000000  libgtk-3-0.dll!gtk_main_do_event
000000006C1CF0AA 0000000000000000 00000000649F4500 00000000019922C0  libgdk-3-0.dll!gdk_event_free
000000006C204D9F 000000000061FB68 0000000000000000 0000000001991340  libgdk-3-0.dll!gdk_win32_drag_context_get_type
00000000649F7BA3 0000000000000002 0000000017B856C0 0000000017B856C0  libglib-2.0-0.dll!g_main_context_dispatch
00000000649F7DD8 0000000000000001 000000000197D930 000000000198CC60  libglib-2.0-0.dll!g_main_context_dispatch
00000000649F8254 0000000000000000 0000000000000000 00000000638FABA0  libglib-2.0-0.dll!g_main_loop_run
000000006785AAAD 0000000000000000 0000000000402CD0 000000000198C750  libgtk-3-0.dll!gtk_main
000000006373217C 00007FFA00000001 000000000197D930 00007FFA00000001  libdarktable.dll!dt_gui_gtk_run  [D:/build/darktable/src/gui/gtk.c @ 1350]
00000000004030AC 00000000004016A0 0000000000000010 0000000000000000  darktable.exe!main  [D:/build/darktable/src/main.c @ 93]
00000000004015F8 000000000000005A 0000000000000000 0000000000408600  darktable.exe!wmain  [D:/build/darktable/src/win/main_wrapper.h @ 15]
00000000004013EE 0000000000000000 0000000000000000 0000000000000000  darktable.exe!__tmainCRTStartup  [D:/mingwbuild/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/crtexe.c @ 334]
000000000040153B 0000000000000000 0000000000000000 0000000000000000  darktable.exe!mainCRTStartup  [D:/mingwbuild/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/crtexe.c @ 223]
00007FFAA0D77BD4 0000000000000000 0000000000000000 0000000000000000  KERNEL32.DLL!BaseThreadInitThunk
00007FFAA0F0CED1 0000000000000000 0000000000000000 0000000000000000  ntdll.dll!RtlUserThreadStart

darktable.exe	3.0.1.0
ntdll.dll   	10.0.18362.719
KERNEL32.DLL	10.0.18362.752
KERNELBASE.dll	10.0.18362.752
msvcrt.dll  	7.0.18362.1
libglib-2.0-0.dll	2.64.0.0
ADVAPI32.dll	10.0.18362.752
sechost.dll 	10.0.18362.693
RPCRT4.dll  	10.0.18362.628
ole32.dll   	10.0.18362.693
combase.dll 	10.0.18362.693
ucrtbase.dll	10.0.18362.387
bcryptPrimitives.dll	10.0.18362.295
GDI32.dll   	10.0.18362.1
win32u.dll  	10.0.18362.752
gdi32full.dll	10.0.18362.719
msvcp_win.dll	10.0.18362.387
USER32.dll  	10.0.18362.719
SHELL32.dll 	10.0.18362.719
cfgmgr32.dll	10.0.18362.387
shcore.dll  	10.0.18362.752
windows.storage.dll	10.0.18362.719
profapi.dll 	10.0.18362.693
powrprof.dll	10.0.18362.1
UMPDC.dll
shlwapi.dll 	10.0.18362.1
kernel.appcore.dll	10.0.18362.1
cryptsp.dll 	10.0.18362.1
WS2_32.dll  	10.0.18362.387
libintl-8.dll	0.19.8.0
libdarktable.dll
PSAPI.DLL   	10.0.18362.1
libiconv-2.dll	1.16.0.0
libpcre-1.dll
libwinpthread-1.dll	1.0.0.0
libgcc_s_seh-1.dll
libcairo-2.dll
libstdc++-6.dll
libcurl-4.dll
CRYPT32.dll 	10.0.18362.592
MSASN1.dll  	10.0.18362.1
WLDAP32.dll 	10.0.18362.449
exchndl.dll 	0.9.2.0
libexiv2.dll
libgdk_pixbuf-2.0-0.dll	2.40.0.0
libgdk-3-0.dll	3.24.14.0
IMM32.dll   	10.0.18362.387
SETUPAPI.dll	10.0.18362.1
bcrypt.dll  	10.0.18362.267
libgmodule-2.0-0.dll	2.64.0.0
libgphoto2-6.dll
libgobject-2.0-0.dll	2.64.0.0
libgphoto2_port-12.dll
libGraphicsMagick-3.dll
libIlmImf-2_4.dll
libjpeg-8.dll
libgtk-3-0.dll	3.24.14.0
libjson-glib-1.0-0.dll
comdlg32.dll	10.0.18362.693
liblcms2-2.dll
lua53.dll
libpango-1.0-0.dll	1.43.0.0
libpangocairo-1.0-0.dll	1.43.0.0
libpng16-16.dll
libpugixml.dll
libsqlite3-0.dll
libsoup-2.4-1.dll
MSIMG32.dll 	10.0.18362.719
libtiff-5.dll
libxml2-2.dll
zlib1.dll
libfreetype-6.dll	2.10.1.0
libfontconfig-1.dll
libbrotlidec.dll
libpixman-1-0.dll
libidn2-0.dll
libpsl-5.dll
libssh2-1.dll
VERSION.dll 	10.0.18362.1
libnghttp2-14.dll
gdiplus.dll 	10.0.18362.752
dwmapi.dll  	10.0.18362.267
WINMM.dll   	10.0.18362.1
libssl-1_1-x64.dll	1.1.1.4
mgwhelp.dll 	0.9.2.0
libcairo-gobject-2.dll
libpangowin32-1.0-0.dll	1.43.0.0
libepoxy-0.dll
libexif-12.dll
libffi-6.dll
libsystre-0.dll
libHalf-2_4.dll
libbz2-1.dll
COMCTL32.dll	6.10.18362.752
WINSPOOL.DRV	10.0.18362.693
libIlmThread-2_4.dll
libatk-1.0-0.dll	2.34.1.0
libpangoft2-1.0-0.dll	1.43.0.0
libharfbuzz-0.dll
libthai-0.dll
liblzma-5.dll	5.2.4.0
libzstd.dll
WINMMBASE.dll	10.0.18362.1
dbghelp.dll 	10.0.18362.1
USP10.dll   	10.0.18362.476
PROPSYS.dll 	7.0.18362.267
OLEAUT32.dll	10.0.18362.693
IPHLPAPI.DLL	10.0.18362.1
libtre-5.dll
dbgcore.DLL 	10.0.18362.1
libgraphite2.dll
libdatrie-1.dll
libgomp-1.dll
libgio-2.0-0.dll	2.64.0.0
libopenjp2-7.dll
librsvg-2-2.dll
libsecret-1-0.dll
libcrypto-1_1-x64.dll	1.1.1.4
libexpat-1.dll
libfribidi-0.dll
libltdl-7.dll
libIex-2_4.dll
libImath-2_4.dll
libbrotlicommon.dll
libunistring-2.dll	0.9.10.0
DNSAPI.dll  	10.0.18362.267
NSI.dll     	10.0.18362.449
libgcrypt-20.dll	1.8.5.54720
libcroco-0.6-3.dll
libgpg-error-0.dll	1.37.0.0
CRYPTBASE.DLL	10.0.18362.1
rsaenh.dll  	10.0.18362.1
uxtheme.dll 	10.0.18362.449
clbcatq.dll 	2001.12.10941.16384
mswsock.dll 	10.0.18362.1
MSCTF.dll   	10.0.18362.752
DEVOBJ.dll  	10.0.18362.387
WINTRUST.dll	10.0.18362.387
Wintab32.dll	1.5.0.0
IconCodecService.dll	10.0.18362.1
WindowsCodecs.dll	10.0.18362.719
winhttp.dll 	10.0.18362.693
libpixbufloader-png.dll
mscms.dll   	10.0.18362.267
USERENV.dll 	10.0.18362.387
ColorAdapterClient.dll	10.0.18362.267
icm32.dll   	10.0.18362.267
TextInputFramework.dll	10.0.18362.693
CoreMessaging.dll	10.0.18362.1
CoreUIComponents.dll	10.0.18362.207
ntmarta.dll 	10.0.18362.1
wintypes.dll	10.0.18362.693
iertutil.dll	11.0.18362.693
OpenCL.dll  	2.2.1.0
nvopencl64.dll	26.21.14.3107
dxcore.dll  	10.0.18362.1
nvapi64.dll 	26.21.14.3107
dxgi.dll    	10.0.18362.693
nvcompiler.dll	26.21.14.3107
nvptxJitCompiler.dll	26.21.14.3107
libdarkroom.dll
libknight.dll
liblighttable.dll
libmap.dll
libosmgpsmap-1.0-1.dll
libslideshow.dll
libtethering.dll
libcopy.dll
libexr.dll
libj2k.dll
libjpeg.dll
libpdf.dll
libpfm.dll
libpng.dll
libppm.dll
libtiff.dll
libwebp.dll
libwebp-7.dll
libdisk.dll
libemail.dll
libgallery.dll
liblatex.dll
libpiwigo.dll
libashift.dll
libatrous.dll
libbasecurve.dll
libbasicadj.dll
libbilat.dll
libbilateral.dll
libbloom.dll
libborders.dll
libcacorrect.dll
libchannelmixer.dll
libclahe.dll
libclipping.dll
libcolisa.dll
libcolorbalance.dll
libcolorchecker.dll
libcolorcontrast.dll
libcolorcorrection.dll
libcolorin.dll
libcolorize.dll
libcolormapping.dll
libcolorout.dll
libcolorreconstruct.dll
libcolortransfer.dll
libcolorzones.dll
libdefringe.dll
libdemosaic.dll
libdenoiseprofile.dll
libdither.dll
libequalizer.dll
libexposure.dll
libfilmic.dll
libfilmicrgb.dll
libfinalscale.dll
libflip.dll
libgamma.dll
libglobaltonemap.dll
libgraduatednd.dll
libgrain.dll
libhazeremoval.dll
libhighlights.dll
libhighpass.dll
libhotpixels.dll
libinvert.dll
liblens.dll
liblensfun.dll
liblevels.dll
libliquify.dll
liblowlight.dll
liblowpass.dll
liblut3d.dll
libmask_manager.dll
libmonochrome.dll
libnlmeans.dll
liboverexposed.dll
libprofile_gamma.dll
librawdenoise.dll
librawoverexposed.dll
librawprepare.dll
librelight.dll
libretouch.dll
librgbcurve.dll
librgblevels.dll
librotatepixels.dll
libscalepixels.dll
libshadhi.dll
libsharpen.dll
libsoften.dll
libsplittoning.dll
libspots.dll
libtemperature.dll
libtonecurve.dll
libtoneequal.dll
libtonemap.dll
libvelvia.dll
libvibrance.dll
libvignette.dll
libwatermark.dll
libzonesystem.dll
libbackgroundjobs.dll
libcamera.dll
libcollect.dll
libcolorlabels.dll
libcolorpicker.dll
libcopy_history.dll
libdarktable_label.dll
libduplicate.dll
libexport.dll
libfilmstrip.dll
libfilter.dll
libgeotagging.dll
libglobal_toolbox.dll
libhinter.dll
libhistogram.dll
libhistory.dll
libimage.dll
libimage_infos.dll
libimport.dll
liblighttable_mode.dll
liblive_view.dll
liblocation.dll
libmap_settings.dll
libmasks.dll
libmetadata.dll
libmetadata_view.dll
libmodulegroups.dll
libmodulelist.dll
libmodule_toolbox.dll
libnavigation.dll
libratings.dll
librecentcollect.dll
libselect.dll
libsession.dll
libsnapshots.dll
libstyles.dll
libtagging.dll
libtimeline.dll
libviewswitcher.dll
libview_toolbox.dll
libpixbufloader-svg.dll

Windows 10.0.18363
DrMingw 0.9.2


```

**Comment by johnny-bit** [comment]
Can you change the report so that instead of  code the log is attached as a file? it's hard to read as a code in github issue.

second - can you export any preset and then import it form 3.0.1 and it happens every time? Can you share the preset file that caused problem?

**Comment by likot180** [comment]
DT 3.4.1.1 crashes when I try to import presets (dtstyles) in darktable preferences (darkroom > show global preferences > presets > import). It's OK when I do it from lighttable > styles > import.

[darktable_bt_3J0NZ0.txt](https://github.com/darktable-org/darktable/files/6052295/darktable_bt_3J0NZ0.txt)
[darktable_bt_31SOZ0.txt](https://github.com/darktable-org/darktable/files/6052297/darktable_bt_31SOZ0.txt)
[darktable_bt_XA2NZ0.txt](https://github.com/darktable-org/darktable/files/6052298/darktable_bt_XA2NZ0.txt)

**Comment by johnny-bit** [comment]
Presets aren't styles though.

**Comment by johnny-bit** [comment]
Main issue fixed in  #7393, issue with importing invalid styles fixed within the scope of #9193
